src/scheduler/sort.cpp

```diff
@@ -280,15 +280,10 @@ cmp_nspec_by_sub_seq(const void *v1, const void *v2)
  * @retval	0 	: if q1 == q2
  * @retval	-1  : if q1 > q2
  */
-int
-cmp_queue_prio_dsc(const void *q1, const void *q2)
+bool
+cmp_queue_prio_dsc (const queue_info *q1, const queue_info *q2)
```

Comment:
What if either are null? Do you want to pass in references instead?

Reply:
Oh i see, it's a vector of pointers. You still might wnat to check for `nullptr` though.

Reply:
These can't be nullptr right? it is called from std::sort which shouldn't be passing a null pointer to sort function.

Reply:
std::sort will pass a reference to the item being sorted, in this case, we're sorting `queue_info *`. Someone can do `vec.push_back(nullptr);`, sort the vector, and this will crash.

Reply:
True, but I think we should be adding such a check when we push_back into vectors. In our case, it is not possible for now, because we are adding queue_info pointers after allocating them and if new fails it is going to raise an exception anyways.

Reply:
That makes sense.

Reply:
remove the space between the function name and the paren.